version: '3.8'

services:
    db:
        image: postgres:16
        container_name: control-tower
        environment:
            POSTGRES_USER: controltower
            POSTGRES_PASSWORD: ct123
            POSTGRES_DB: control-tower-db
        ports:
            - "5434:5432"
        healthcheck:
            test: ['CMD-SHELL', 'pg_isready -U controltower -d control-tower-db']
            interval: 1s
            timeout: 1s
            retries: 15
        volumes:
            - postgres_data:/var/lib/postgresql/data
    db-test:
        image: postgres:16
        container_name: control-tower-test
        environment:
            POSTGRES_USER: controltower
            POSTGRES_PASSWORD: ct123
            POSTGRES_DB: control-tower-db-test
        ports:
            - "5435:5432"
        volumes:
            - db_test:/var/lib/postgresql/data

    api:
        build:
            context: .
            dockerfile: dockerfile
        container_name: api
        restart: always
        environment:
            - DATABASE_URL=postgresql://controltower:ct123@db:5432/control-tower-db
        env_file:
            - .env
        ports:
            - '3001:3001'
        depends_on:
            db:
                condition: service_healthy
        links:
            - db
        deploy:
            resources:
                limits:
                    cpus: '1'
                    memory: 4000M
        extra_hosts:
            - "host.docker.internal:host-gateway"

volumes:
    postgres_data:
    db_test:

